nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -29,28 +34,54 @@ public static NsisExecutable createNsisExecutable(GenericFactory factory,
 	private void initNsisExecutable(GenericFactory factory, ByteProvider bp,
 			SectionLayout layout) throws IOException {
 		this.reader = new FactoryBundledWithBinaryReader(factory, bp, true);
-		this.header_offset = -1;
+		this.headerOffset = -1;
 		findHeaderOffset();
-		// TODO Init sections and headers
+		initScriptHeader(bp);
 	}
 
 	private void findHeaderOffset() throws IOException {
-		if (this.header_offset == -1) {
+		if (this.headerOffset == -1) {
 			for (long offset = 0; offset
 					+ NsisConstants.NSIS_MAGIC.length <= reader
 							.length(); offset++) {
 				byte[] content = reader.readByteArray(offset,
 						NsisConstants.NSIS_MAGIC.length);
 				if (Arrays.equals(NsisConstants.NSIS_MAGIC, content)) {
-					this.header_offset = offset;
+					this.headerOffset = offset;
 					return;
 				}
 			}
 		}
 	}
 
+	private void initScriptHeader(ByteProvider bp) throws IOException {
+		BinaryReader br = new BinaryReader(bp, true);
+		br.setPointerIndex(this.headerOffset);
+		this.scriptHeader = new NsisScriptHeader(br);
+	}
+
 	public long getHeaderOffset() {
-		return this.header_offset;
+		return this.headerOffset;
+	}
+
+	public int getInflatedHeaderSize() {
```

Comment:
For this function and the others that get directly from the header object.

What do you think about letting the caller call "getHeader()" and return the NsisHeader object, the caller can then get the information of the header object it wants.

I suggest this because it seems to be how Ghidra implemented it for PortableExecutable:
https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/pe/PortableExecutable.java

Reply:
I thought of it. I wrote it that way because I was considering the [law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter). It makes the code easier to understand/maintain in a way, since NsisLoader does not have to know that NsisHeader object is a parameter of NsisExecutable. What are your thoughts on that?

Reply:
Understood, I agree that returning a Header object does violate the Demeter Law.

I am questioning the need for `NsisHeader` at all, why not just have one class `NsisExecutable`?

What about this:
If `NsisHeader` is useful because it encapsulates logic could it be an inner class of `NsisExectuable` and the data members of `NsisHeader` are directly available to NsisExectuable?

So NsisExecutable.getInflatedHeaderSize becomes:
```java
public int getInflatedHeaderSize() {
  return this.scriptHeader.inflatedHeaderSize;
}
```

And `NsisHeader.getInflatedHeaderSize()` and all the other getters and setters just don't exist anymore, what do you think?

I would like to get rid of getters and setters that just call other ones.

Reply:
I really like that! It's way cleaner

Reply:
Changed